### PR TITLE
Set local_devices from CUDA_VISIBLE_DEVICES in initialize_jax_for_gpu

### PR DESCRIPTION
**Pull request by gabeweisz**
# Description

When running in Slurm, jax.init_distributed.initialize only attaches to one GPU unless local_device_ids is passed.
This PR uses the standard CUDA_VISIBLE_DEVICES environment variable to set local_device_ids,
falling back to the default behavior if CUDA_VISIBLE_DEVICES is None or cannot be parsed.

We have been using this internally at AMD for some time.

This change will not affect TPUs in any way.

FIXES: #865 

# Tests

Run a job in Slurm on a machine with multiple GPUs with and without CUDA_VISIBLE_DEVICES set

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Comment by gabeweisz** [comment]
I just updated from head, would appreciate a review

**Review by shralex — APPROVED** [review]

**Comment by gabeweisz** [comment]
I will address these issues - thank you for the feedback.

File: src/maxtext/utils/max_utils.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-""" Common Max Utils needed by multiple modules.
+"""Common Max Utils needed by multiple modules.
```
Comment (gabeweisz) [inline src/maxtext/utils/max_utils.py:15]: pylint made this change

File: tests/unit/max_utils_test.py [hunk 2]
```diff
@@ -12,12 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-""" Tests for the common Max Utils """
+"""Tests for the common Max Utils"""
```
Comment (gabeweisz) [inline tests/unit/max_utils_test.py:15]: pylint made this change

**Comment by gabeweisz** [comment]
Tests added. I'll make a follow up PR in a few days with some documentation about how to use Slurm with MaxText.

All of the failing code quality checks are in [files that I did not touch](https://github.com/AI-Hypercomputer/maxtext/actions/runs/22366290074/job/64733071073?pr=2945)

Can I have another review? What should I do about the files that I did not touch that fail?

**Review by dipannita08 — APPROVED** [review]
Thank you!

**Comment by gabeweisz** [comment]
All tests have passed, and I have the required approvals, but do not have the right github permissions to merge this. Can someone who has permission to merge this PR please do so?
